_source/_posts/2022-01-13-node-deploy-to-heroku.md

```diff
@@ -0,0 +1,492 @@
+---
+layout: blog_post
+title: ""
+author:
+by: advocate|contractor
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness|conversion
+---
+
+## Introduction
+
+Heroku is a platform as a service that supports many languages.  Initially, it was developed to support Ruby sites but now supports an array of languages including javascript with Node.js.  Heroku also has docker support so you can deploy just about anything to it.  
+
+However, in this tutorial, you will learn how to build a small application using the `express` framework for Node.js.  You will then secure that application using Okta by creating an application in Okta for web applications and integrating the Okta OIDC middleware with your application.  Finally, you will learn how to deploy the application to Heroku, update your Okta application for your Heroku site, and how to set the environment variables for your app in Heroku.
+
+To continue you will need:
+
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [Node.Js](https://nodejs.org/en/)
+- An [Okta Developer Account](https://developer.okta.com/) (free forever, to handle your OAuth needs)
+- A [Heroku Account](https://signup.heroku.com/)
+- A [Git Repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)
+
+Of course, if you want to just see the code feel free [to view it on github](https://github.com/nickolasfisher/Okta_Heroku).
+
+## Create your Okta Application
+
+{% include setup/cli.md type="web" framework="node"
+   loginRedirectUri="http://localhost:3000/authorization-code/callback" signup="false" %}
+
+## Create your Express Application
+
+Now it's time to turn your attention to writing your application.   YOu will quickly scaffold your application by using the `express application generator` with the npx command `npx express-generator`.  Open the folder where your application will live and run the command `npx express-generator`.  
+
+### Install your Dependencies
+
+Next, run the command to install the default dependencies from express-generator.
```

Comment:
Does it make sense to give a bullet list with an explanation of what each package does and have all the commands in a single block? It will make it easier for readers to copy and paste.

Reply:
That's your call.  I find it more difficult to read so I don't write it that way.  If you'd rather thats okay